storage/src/testFixtures/java/tech/pegasys/teku/storage/client/ChainUpdater.java

```diff
@@ -166,6 +173,14 @@ public void syncWith(final ChainBuilder otherChain) {
     updateBestBlock(otherChain.getLatestBlockAndState());
   }
 
+  public void syncWithUpToSlot(final ChainBuilder otherChain, final long slot) {
+    otherChain
+        .streamBlocksAndStates()
+        .filter(signedBlockAndState -> signedBlockAndState.getSlot().isLessThanOrEqualTo(slot))
```

Comment:
nit: Could just use:
```suggestion
        .streamBlocksAndStates(0, slot)
```